resources/pdns_authoritative_config.rb

```diff
@@ -71,12 +72,15 @@
   directory new_resource.socket_dir do
     owner new_resource.run_user
     group new_resource.run_group
-    mode '0755'
+    # Because of the DynListener creation before dropping privileges, the
+    # socket-directory has to be '0777' for now
+    # Issue: https://github.com/PowerDNS/pdns/issues/4826
+    mode Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd) ? '0777' : '0755'
     recursive true
     action :create
   end
 
-  template "#{new_resource.config_dir}/pdns-authoritative-#{new_resource.instance_name}.conf" do
+  template "#{new_resource.config_dir}/pdns-#{new_resource.instance_name}.conf" do
```

Comment:
I agree, the only issue is that `pdns` when used with `--config-name=INSTANCE_NAME` looks for `pdns-INSTANCE_NAME`. But I will double check.

The pattern I tried to use is the [virtual-hosting](https://github.com/PowerDNS/pdns/blob/master/docs/markdown/authoritative/running.md#virtual-hosting) feature of pdns without drifting too much. Is that a direction you want to go to?